### ISSUE TITLE
Bank account - Green box for completed part is clickable but nothing happens when clicking on it

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#36506]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.41-2
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to workspace settings > any workspace > Bank account.
3. Click Connect manually.
4. Finish Bank info and proceed to Personal info.
5. Click on the green box with checkmark.

## Expected Result:
Since the green box is clickable (with hand cursor), clicking on it should navigate to the first section.

## Actual Result:
The green box is clickable, but nothing happens when clicking on it.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/f7177ccf-e6d6-4882-bbfd-9580b1555beb

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
We think that this bug might be related #wave5-free-submitters
CC @dylanexpensify

**Comment by mountiny** [comment]
@paultsimura yep there are cases when we want to lock the steps https://github.com/Expensify/App/pull/36472/files#diff-50cbf33e1a4add301c4c85b2d37735fa79094c511c6fc0fe0003940870ec4a1dR81

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?
The steps are pressable, but they shouldn't be is there's no `onStepSelected` passed.

### What changes do you think we should make in order to solve the problem?
Use `!onStepSelected && styles.cursorDefault,` to not modify the cursor at all:

https://github.com/Expensify/App/blob/c03a1a2a28caeab970c2aa447a87e7cb7fe4613f/src/components/InteractiveStepSubHeader.tsx#L76-L80

### What alternative solutions did you explore? (Optional)

**Comment by paultsimura** [comment]
@akinwale with all due respect, please note the following:
1. The solution by @FitseTLT is already in `main` – that's what @mountiny referred in the linked PR, and I mentioned it in my proposal – so if you choose that proposal, we will just do nothing here. But as I mentioned in my proposal, with this approach, the cursor will always show "forbidden" and it's not the best UX:

> After https://github.com/Expensify/App/pull/36472/files#diff-50cbf33e1a4add301c4c85b2d37735fa79094c511c6fc0fe0003940870ec4a1dR81, the cursor will always be "disabled" instead of the "pointer".

Code in main:
https://github.com/Expensify/App/blob/f386740fb579a628c845745189a03aa55c77e3bf/src/components/InteractiveStepSubHeader.tsx#L81

2. My changes (after other proposals were posted) were mostly cosmetic/formatting – I do post the Updated Proposal if I change some useful content in the proposal and there are others. IMO, here it was not needed.

I still think there isn't a better proposal than mine, and I'll just repost it so you can re-check ignoring the editing timestamps.

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The cursor on steps is "pointer" even if they are not clickable. After https://github.com/Expensify/App/pull/36472/files#diff-50cbf33e1a4add301c4c85b2d37735fa79094c511c6fc0fe0003940870ec4a1dR81, the cursor will always be "disabled" instead of the "pointer".

### What is the root cause of that problem?
The steps are pressable, but they shouldn't be if there's no `onStepSelected` passed as discussed in https://github.com/Expensify/App/issues/36506#issuecomment-1944046215.

### What changes do you think we should make in order to solve the problem?
Use `!onStepSelected && styles.cursorDefault,` to not modify the cursor at all and use the default one if the steps are not meant to be clickable:

https://github.com/Expensify/App/blob/c03a1a2a28caeab970c2aa447a87e7cb7fe4613f/src/components/InteractiveStepSubHeader.tsx#L76-L80

### What alternative solutions did you explore? (Optional)

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The cursor on steps is "forbidden" even if they are not clickable by design.

### What is the root cause of that problem?
The steps are pressable but disabled, which results in the "forbidden" cursor being displayed. They shouldn't be treated as pressable at all if there's no `onStepSelected` passed as discussed in https://github.com/Expensify/App/issues/36506#issuecomment-1944046215.

### What changes do you think we should make in order to solve the problem?
Use `!onStepSelected && styles.cursorDefault,` to not modify the cursor at all and use the default one if the steps are not meant to be clickable:

https://github.com/Expensify/App/blob/c03a1a2a28caeab970c2aa447a87e7cb7fe4613f/src/components/InteractiveStepSubHeader.tsx#L76-L80

### What alternative solutions did you explore? (Optional)

**Comment by akinwale** [comment]
@paultsimura Thanks for the clarification. Could you update your proposal with what the updated style code would look like?

**Comment by FitseTLT** [comment]
By the way the code change in `main` was made after my proposal so in my opinion we ain't need any change as if we don't have nothing to do on pressing (early returning with that condition) we should make it disabled with same condition

**Comment by akinwale** [comment]
After a second review, since the only change required for this issue is the cursor style update, we can move forward with @paultsimura's [proposal](https://github.com/Expensify/App/issues/36506#issuecomment-1945563614).

cc @nkuoch

**Comment by paultsimura** [comment]
> @paultsimura can you raise a PR please?

Sure, on it now

**Comment by paultsimura** [comment]
Thanks. The PR is ready for review: https://github.com/Expensify/App/pull/36573

**Comment by paultsimura** [comment]
Deployed to prod: https://github.com/Expensify/App/pull/36573#issuecomment-1965250260

**Comment by mountiny** [comment]
Payment for @akinwale will be handled here https://github.com/Expensify/App/issues/22901#issuecomment-1977046291

**Comment by bfitzexpensify** [comment]
Payment for @paultsimura complete, closing this out.
